### PR TITLE
adapter: improve schema resolution

### PR DESCRIPTION
**Pull request by mjibson**
Improve schema resolution:

- Correct `current_schema[s]()` implementations.
- Use the session's `search_path` to determine the schema in which to create new objects.

See #6754

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

File: src/sql/src/plan/error.rs [hunk 1]
```diff
@@ -138,6 +138,9 @@ pub enum PlanError {
         name: String,
         supported_azs: BTreeSet<String>,
     },
+    // TODO: This should have code 3F000, but there's no place to specify that
+    // yet. Maybe this needs to be a CatalogError?
```
Comment (jkosh44) [inline src/sql/src/plan/error.rs:142]: Should we create a GH issue for this?

File: src/adapter/src/catalog.rs [hunk 2]
```diff
@@ -1303,7 +1303,7 @@ impl ConnCatalog<'_> {
         }
     }
 
-    fn effective_search_path(
+    pub fn effective_search_path(
```
Comment (jkosh44) [inline src/adapter/src/catalog.rs:1306]: I think a comment on what this function does would be helpful.

**Review by jkosh44 — APPROVED** [review]
LGTM

File: src/sql/src/plan/error.rs (hunk 1 above)
Comment (mjibson) [reply src/sql/src/plan/error.rs]: I figured out how to do this instead!

**Pull request merged**